### PR TITLE
Update getInstallPath() to return an absolute path

### PR DESCRIPTION
**Pull request by joshdaws**
This is to address an issue preventing path repos from working with
Installers. They require an absolute path for creating a symlink

### CODE REVIEW

**Comment by slbmeh** [comment]
The `getcwd` command isn't quite what you want here. It works in the most basic scenario but some CI/CD configurations might not be in a working directory of the project.

**Comment by joshdaws** [comment]
Thanks, @slbmeh I've modified my PR to check for absolute paths.  Please have another look.  I've tested this with a number of installer-path configurations:

../themes/{$name}
web/app/themes{$name}
/Users/username/www/site/web/app/themes/{$name} (absolute path to current project)
/User/username/Desktop/themes/{$name} (absolute path outside of current project)

All work as expected for Path repositories (installed via symlink) and VCS repositories (installed normally)

**Comment by thelostspore** [comment]
@slbmeh can you please take another look - would be great if this project worked well with composer path repositories.

File: src/Composer/Installers/BaseInstaller.php [hunk 1]
```diff
@@ -111,10 +111,29 @@ protected function templatePath($path, array $vars = array())
...
+     * @param  string $path
+     * @return boolean 
+     */
+    protected function isAbsolutePath($path)
+    {
+        if ($path === null || $path === '') {
+            return false;
+        }
+
+        return $path[0] === DIRECTORY_SEPARATOR || preg_match('~\A[A-Z]:(?![^/\\\\])~i',$path) > 0;
```
Comment (Seldaek) [inline src/Composer/Installers/BaseInstaller.php]: You can use Composer\Util\Filesystem->isAbsolutePath() instead of reimplementing this here.

**Comment by mglaman** [comment]
@joshdaws do you want a PR to your fork for feedback? Currently blocking a project setup, unless I use `cweagans/composer-patches` to patch from PR.

**Comment by joshdaws** [comment]
@mglaman Sure. I've been swamped at work and haven't been able to address this.

**Comment by dzuelke** [comment]
The symlink should not be absolute, since you then cannot move the built package to another machine with a different directory structure.

I am encountering a similar problem to this or #282, but the problem looks to be somewhere else.

**Comment by dzuelke** [comment]
Okay, figured out what the underlying problem really is: trailing slashes in your installer paths. If you remove them, the symlink succeeds.

**Comment by dzuelke** [comment]
https://github.com/composer/composer/pull/5164 is the correct fix for this!

**Comment by Seldaek** [comment]
I merged https://github.com/composer/composer/pull/5164 - can someone confirm that it fixes the issue? You just need to `composer self-update --snapshot` to try the latest code.

**Comment by Seldaek** [comment]
Going to close assuming @dzuelke is correct but if the error is still present feel free to nag :)

**Pull request closed without merging**